presto-main/src/main/java/com/facebook/presto/type/DecimalInequalityOperators.java

```diff
@@ -218,4 +220,167 @@ else if (aType instanceof LongDecimalType && bType instanceof LongDecimalType) {
+            else if (vType instanceof ShortDecimalType && aType instanceof LongDecimalType && bType instanceof LongDecimalType) {
+                baseMethodHandle = OP_SHORT_LONG_LONG;
+            }
+            else if (vType instanceof LongDecimalType && aType instanceof ShortDecimalType && bType instanceof ShortDecimalType) {
+                baseMethodHandle = OP_LONG_SHORT_SHORT;
+            }
+            else if (vType instanceof LongDecimalType && aType instanceof ShortDecimalType && bType instanceof LongDecimalType) {
+                baseMethodHandle = OP_LONG_SHORT_LONG;
+            }
+            else if (vType instanceof LongDecimalType && aType instanceof LongDecimalType && bType instanceof ShortDecimalType) {
+                baseMethodHandle = OP_LONG_LONG_SHORT;
+            }
+            else if (vType instanceof LongDecimalType && aType instanceof LongDecimalType && bType instanceof LongDecimalType) {
+                baseMethodHandle = OP_LONG_LONG_LONG;
+            }
+            else {
+                throw new IllegalArgumentException(format("unsuported argument types (%s, %s, %s)", vType, aType, bType));
+            }
+
+            MethodHandle lowerBoundTestMethodHandle = DECIMAL_LESS_THAN_OR_EQUAL_OPERATOR.specialize(ImmutableMap.<String, Type>of("A", aType, "B", vType), 2, typeManager, functionRegistry).getMethodHandle();
+            MethodHandle upperBoundTestMethodHandle = DECIMAL_GREATER_THAN_OR_EQUAL_OPERATOR.specialize(ImmutableMap.<String, Type>of("A", bType, "B", vType), 2, typeManager, functionRegistry).getMethodHandle();
+
+            MethodHandle methodHandle = MethodHandles.insertArguments(baseMethodHandle, 3, lowerBoundTestMethodHandle, upperBoundTestMethodHandle);
+            return operatorInfo(BETWEEN, BOOLEAN_SIGNATURE, ImmutableList.of(vType.getTypeSignature(), aType.getTypeSignature(), bType.getTypeSignature()),
+                    methodHandle, false, ImmutableList.of(false, false, false));
+        }
+
+        public static boolean betweenShortShortShort(long v, long a, long b, MethodHandle lowerBoundTestMethodHandle, MethodHandle upperBoundTestMethodHandle)
+        {
+            try {
+                return (boolean) lowerBoundTestMethodHandle.invokeExact(a, v) && (boolean) upperBoundTestMethodHandle.invokeExact(b, v);
+            }
+            catch (Throwable t) {
+                Throwables.propagateIfInstanceOf(t, Error.class);
+                Throwables.propagateIfInstanceOf(t, PrestoException.class);
+                throw new PrestoException(INTERNAL_ERROR, t);
+            }
+        }
+
+        public static boolean betweenShortShortLong(long v, long a, Slice b, MethodHandle lowerBoundTestMethodHandle, MethodHandle upperBoundTestMethodHandle)
+        {
+            try {
+                return (boolean) lowerBoundTestMethodHandle.invokeExact(a, v) && (boolean) upperBoundTestMethodHandle.invokeExact(b, v);
+            }
+            catch (Throwable t) {
```

Comment:
use the same schema as with previous patch in order to have catch(Throwable t)  in one place